android: Switch to the LLVM libc++ runtime

cnorthrop opened the pull request:
This includes support for std::to_string, so we can also remove the overrides.

Requires switching our shaderc build over too, which is done with a custom Application.mk.

krOoze commented:
Google seems to consider it not stable enough yet.
https://developer.android.com/ndk/guides/cpp-support.html
"The NDK's libc++ is not stable. Not all the tests pass, and the test suite is not comprehensive."

While messing with this, would it make sense to also try switch `-std=c++14`?

Why is shaderc `android-23` while the rest is `android-22`?

mark-lunarg commented:
@cnorthrop, is this one toast?  Anything we can do to move it along, or should we close this MR for now?

cnorthrop commented:
@stroyan has done the most recent work here and can let us know if we should close this one or replace it.

stroyan commented:
Times have changed.  As of NDK r16 libc++ is the preferred STL.
In the next release it will be the default STL.
The time has come to take this PR.

https://android.googlesource.com/platform/ndk/+/ndk-release-r16/CHANGELOG.md

> 
libc++ is out of beta and is now the preferred STL in the NDK. Starting in r17, libc++ is the default STL for CMake and standalone toolchains. If you manually selected a different STL, we strongly encourage you to move to libc++.

tobine commented:
@cnorthrop can you rebase off latest master?

krOoze commented:
@stroyan The CI servers are at r13 though, where it is still beta. Is there any reason to use old NDK, or should it be updated here?

cnorthrop commented:
Okay, rebased to master, also updated the Travis config.  We'll need to update our respective test farms and downstream repos to NDK r16b.  I'll do that on our end now.

Tony-LunarG commented:
Ok, I've updated our farm system - LGTM

cnorthrop commented:
Test run failed internally here.  Both packaging problems and new test failures.  Will work with Tony to get the packing fixed first, then focus on new test failures.

cnorthrop commented:
Farms are updated, packaging working.

Seeing some failures when running VkLayerValidationTests executable on rooted devices when running more than 23 tests.  Since that doesn't affect most users and running individual tests is fine, let's move forward with this PR.  We can use r15c on our farm until we have a chance to root cause the issue, which will be a high priority.

cnorthrop commented:
cherry-picked to master
